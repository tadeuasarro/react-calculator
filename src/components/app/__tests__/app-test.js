import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../app';

describe('Display', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('creates a Home component', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
