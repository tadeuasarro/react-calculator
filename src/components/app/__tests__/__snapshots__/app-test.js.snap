// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<main>
  <div
    id="navbar"
  >
    <h1>
      Math-magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/app"
          onClick={[Function]}
        >
          App
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    id="outer-container"
  >
    <div
      id="container"
    >
      <div
        id="display"
      />
      <div>
        <div
          className="row"
          id="operations"
        >
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="button-orange button-shape"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="button-orange button-shape"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="button-orange button-shape"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="button-orange button-shape"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
        <div
          className="row"
        >
          <button
            className="button-gray button-zero"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="button-gray button-shape"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="button-orange button-shape"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
